### PR TITLE
Create scheduled task to send inactive users security notification

### PR DESCRIPTION
**Pull request by michalkleiner**
### Description:

- Add a scheduled task to run once a month
- Add email templates for inactive users security notification
- Add notification providers and other classes similar to how token notifications are handled
- Add integration tests

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: plugins/UsersManager/Emails/InactiveUsersNotificationEmail.php [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+     */
+    private $notification;
+
+    /** @var string */
+    private $recipient;
+
+    /** @var array */
+    private $emailData;
+
+    private $userModel;
```
Comment (caddoo) [inline plugins/UsersManager/Emails/InactiveUsersNotificationEmail.php]: Small detail, can you make the comment styling consistent and add one for $userModel.

File: plugins/UsersManager/UserNotifications/InactiveUsersEmailNotification.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\Plugins\UsersManager\UserNotifications;
+
+use Piwik\Plugins\UsersManager\Emails\InactiveUsersNotificationEmail;
+
+final class InactiveUsersEmailNotification extends UserEmailNotification
+{
+    public function __construct(
```
Comment (caddoo) [inline plugins/UsersManager/UserNotifications/InactiveUsersEmailNotification.php:21]: Can we add a doc block around what the params are.

File: plugins/UsersManager/UserNotifications/UserNotifierTask.php [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    /**
+     * Dispatch notifications for each provider and its users
+     */
+    public function dispatchNotifications()
+    {
+        $container = StaticContainer::getContainer();
+        $logger = $container->get(LoggerInterface::class);
+
+        try {
+            Option::set(self::LAST_RUN_TIME_OPTION_NAME, Date::factory('today')->getTimestamp());
```
Comment (caddoo) [inline plugins/UsersManager/UserNotifications/UserNotifierTask.php:59]: Should this be set at the end not at the start?

File: plugins/UsersManager/tests/Integration/InactiveUsersNotificationEmailTest.php [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+use Piwik\Tests\Framework\Mock\FakeAccess;
+use Piwik\Tests\Framework\TestCase\IntegrationTestCase;
+
+/**
+ * @group UsersManager
+ * @group InactiveUsersNotificationEmailTest
+ * @group UserNotifications
+ * @group Plugins
+ */
+class InactiveUsersNotificationEmailTest extends IntegrationTestCase
```
Comment (caddoo) [inline plugins/UsersManager/tests/Integration/InactiveUsersNotificationEmailTest.php:29]: Nice clean integration test 👍

File: plugins/UsersManager/UserNotifications/UserNotifierTask.php (hunk 3 above)
Comment (michalkleiner) [reply plugins/UsersManager/UserNotifications/UserNotifierTask.php:59]: We have it at the beginning for the token notifier task as well as the GeoIP updater task, so it's following the same pattern. I think it's ok at the start in case anything failed during the processing we would have it recorded when it ran (plus the exception logged in the logs).

File: plugins/UsersManager/UserNotifications/InactiveUsersEmailNotification.php (hunk 2 above)
Comment (michalkleiner) [reply plugins/UsersManager/UserNotifications/InactiveUsersEmailNotification.php:21]: Added here and in the abstract parent as well.

File: plugins/UsersManager/Emails/InactiveUsersNotificationEmail.php (hunk 1 above)
Comment (michalkleiner) [reply plugins/UsersManager/Emails/InactiveUsersNotificationEmail.php]: Added, good spotting.

**Review by caddoo — APPROVED** [review]
